### PR TITLE
Create testPlan.md

### PR DESCRIPTION
**Pull request by siberden**
Refined test plan with detailed scope, testing methodologies (manual and automated), and key deliverables. Ensures robust validation of UI components and workflows. Success criteria defined to guarantee seamless functionality and defect resolution.

## Description

_Give a brief context and description of the code change. Highlight anything you want reviewers to pay attention to._

_The PR should have enough information so someone coming in fresh gets a sense of what's going on, but the details of the overall project can be documented on project.digitalaidseattle.org_

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

_Link user story from projects.digitalaidseattle.org_

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well as any relevant images for UI changes._

### CODE REVIEW

File: docs/testPlan.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+## Test Plan for Inventory Management System
+
+### 1. Objective
+The purpose of this test plan is to validate the UI functionality of the inventory management system, ensuring that all components work as expected, the user experience is smooth, and workflows are free of errors. The focus is on performing UI end-to-end (E2E) testing for core features.
+
+### 2. Scope
+**Included:**
+- Home page: Add and checkout inventory functionality.
+- Inventory page: Filtering and displaying inventory.
+- Checkout page: Adding items to checkout and cart management.
```
Comment (bospoort) [inline docs/testPlan.md:15]:
```suggestion
- Checkout page: Adding items to checkout and cart management.
- People page: Adding new volunteers, deactivating roles, view PIN
```

File: docs/testPlan.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+| Tester     | Execute test cases and log defects. |
+| Developer  | Resolve bugs identified.          |
+
+### 6. Test Environment
+- **Browsers:** Chrome, Firefox, and Edge (latest versions).
+- **Devices:** Desktop and mobile (responsive testing).
+- **Test Data:** Realistic inventory dataset with various types, categories, and statuses.
+
+### 7. Test Cases
+#### Home Page
```
Comment (bospoort) [inline docs/testPlan.md:44]: We should also have a section on the Volunteer selector page.

File: docs/testPlan.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+- Verify the "+" button’s visibility and functionality.
+- Test the new window for proper rendering of fields: inventory type, item name, quantity, "Cancel," and "Add" buttons.
+- Validate field input rules (e.g., quantity accepts only numbers).
+- Confirm "Add" button functionality adds inventory successfully.
+
+#### Inventory Page
+- Test inventory items’ display accuracy.
+- Verify type, category, and status filters (individually and combined).
+- Test scenarios with no matching results for applied filters.
```
Comment (bospoort) [inline docs/testPlan.md:53]: we should also verify that Add from this page works.

**Review by bospoort — CHANGES_REQUESTED** [review]
Looks good. Some additional things we can test are added.

File: docs/testPlan.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+### 1. Objective
+The purpose of this test plan is to validate the UI functionality of the inventory management system, ensuring that all components work as expected, the user experience is smooth, and workflows are free of errors. The focus is on performing UI end-to-end (E2E) testing for core features.
+
+### 2. Scope
+**Included:**
+- Home page: Add and checkout inventory functionality.
+- Inventory page: Filtering and displaying inventory.
+- Checkout page: Adding items to checkout and cart management.
+- People page: Adding new volunteers, deactivating roles, view PIN
+
```
Comment (siberden) [inline docs/testPlan.md:17]:
```suggestion
- Volunteer Selector Page: Adding new volunteers, viewing roles, deactivating volunteers, and verifying the "Add" functionality. 
```

File: docs/testPlan.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+**Excluded:**
```
Comment (siberden) [inline docs/testPlan.md:18]:
```suggestion
Volunteer Switching: Validating the workflow for switching volunteers between roles.
```

File: docs/testPlan.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+- Integration testing with backend services.
```
Comment (siberden) [inline docs/testPlan.md:19]:
```suggestion
**Excluded:**
- Integration testing with backend services.
```

File: docs/testPlan.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+- **Devices:** Desktop and mobile (responsive testing).
+- **Test Data:** Realistic inventory dataset with various types, categories, and statuses.
+
+### 7. Test Cases
+#### Home Page
+- Verify the "+" button’s visibility and functionality.
+- Test the new window for proper rendering of fields: inventory type, item name, quantity, "Cancel," and "Add" buttons.
+- Validate field input rules (e.g., quantity accepts only numbers).
+- Confirm "Add" button functionality adds inventory successfully.
+
```
Comment (siberden) [inline docs/testPlan.md:50]:
```suggestion
#### Volunteer Selector Page
- Verify the form for adding new volunteers is rendered correctly and validates input fields.
- Test successful submission of new volunteers.
- Validate functionality for deactivating volunteer roles.
- Ensure roles are accurately displayed and updated in the system.
- Confirm the "Add" functionality works as expected.
```

File: docs/testPlan.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+#### Inventory Page
+- Test inventory items’ display accuracy.
+- Verify type, category, and status filters (individually and combined).
+- Test scenarios with no matching results for applied filters.
+
+#### Checkout Page
+- Ensure "+" button adds items to the cart.
+- Validate the 10-item limit for the cart.
+- Test edge cases, such as removing items and exceeding the limit.
+
```
Comment (siberden) [inline docs/testPlan.md:60]:
```suggestion
#### People Page
- Test adding new volunteers and verifying their roles.
- Validate proper management of roles, including activation and deactivation.
- Confirm the PIN functionality displays securely and accurately.

#### Volunteer Switching
- Verify the workflow for switching volunteers between roles is seamless and error-free.
- Validate that updated roles reflect accurately across the system.
```

File: docs/testPlan.md (hunk 2 above)
Comment (siberden) [reply docs/testPlan.md:57]: Volunteer selector page added

**Review by bospoort — APPROVED** [review]

**Review by mistyb01 — APPROVED** [review]
Looks good, thanks!

**Comment by DrewLovell** [comment]
Looks good! Interested in the schedule and how that evolves as we follow a more structured testing format.
